src/neo-vm/Types/Buffer.cs

```diff
@@ -24,34 +25,47 @@ public class Buffer : StackItem
         /// <summary>
         /// The internal byte array used to store the actual data.
         /// </summary>
-        public readonly byte[] InnerBuffer;
+        public readonly Memory<byte> InnerBuffer;
 
         /// <summary>
         /// The size of the buffer.
         /// </summary>
         public int Size => InnerBuffer.Length;
         public override StackItemType Type => StackItemType.Buffer;
 
+        private readonly byte[] _buffer;
+        private bool _keep_alive = false;
+
         /// <summary>
         /// Create a buffer of the specified size.
         /// </summary>
         /// <param name="size">The size of this buffer.</param>
         /// <param name="zeroInitialize">Indicates whether the created buffer is zero-initialized.</param>
         public Buffer(int size, bool zeroInitialize = true)
         {
-            InnerBuffer = zeroInitialize
-                ? new byte[size]
-                : GC.AllocateUninitializedArray<byte>(size);
+            _buffer = ArrayPool<byte>.Shared.Rent(size);
+            InnerBuffer = new Memory<byte>(_buffer, 0, size);
```

Comment:
I find `_buffer.AsMemory` a little clearer, though `new Memory` also works

Reply:
`new Memory` is invoked inside `AsMemory` too. So, we'd better `new` directly.